8.x/theme-development/helpers.md

```diff
@@ -10,424 +10,1608 @@ First, open the helper file: `app/Utils/CMS/Template/helpers.php`.
+__________________________
+
+
 
+### recaptcha_enabled
 
-### recaptcha_enabled()
 This function determines whether Recaptcha is enabled on the system or not.
+__________________________
+
+
 
+### get_same_models_products
 
-### get_same_models_products()
 Takes a product from the input and returns same model products. For example: Different sizes of a shoe are same model products.
+__________________________
+
+
+
+### check_cart
 
-### check_cart()
 Checks if the product is in the cart. It is suggested not to use this function due to low efficiency.
+__________________________
+
+
 
+### get_cart_information
 
-### get_cart_information()
 This function returns cartRows data related to a product_id.
+__________________________
 
 
-### get_cart()
+
+
+### get_cart
+
 This function returns the structured cart of the current customer whether he is logged in or not.
+__________________________
+
+
 
 
-### get_minimum_purchase_free_shipment()
+### get_minimum_purchase_free_shipment
+
 This function specifies how much to buy to make shipping free.
+__________________________
+
+
 
 
-### product_disable_on_min()
+### product_disable_on_min
+
 This function specifies when a product is disabled, when it is zero, or when it reaches a minimum.
+__________________________
+
+
 
 
-### customer_can_edit_profile()
+### customer_can_edit_profile
+
 This function specifies whether the customer is allowed to edit their profile or not.
+__________________________
+
+
 
+### get_root_directory_per_directory
 
-### get_root_directory_per_directory()
 This function takes a directory and returns root node.
+__________________________
+
+
 
+### h_view
 
-### h_view()
 This function must move to the common helpers; It's not related to the template engine.
+__________________________
 
 
-### get_cms_setting()
+
+### get_cms_setting
+
 This function takes the key from the input and returns the value set in the database by the administrator of e-commerce.
 For example if you call get_cms_setting('phone_number') it will return the value passed for phone_number by the administrator.
+__________________________
+
 
 
-### get_template_views()
+### get_template_views
+
 This function returns a list of available template original views' names.
+__________________________
+
+
 
+### get_current_customer_location
 
-### get_current_customer_location()
 There is no explanation for this function.
+__________________________
 
 
-### get_current_customer_location_data()
+
+### get_current_customer_location_data
+
 There is no explanation for this function.
+__________________________
+
 
 
-### get_customer_meta_categories()
+### get_customer_meta_categories
+
 This function returns a list of available customer meta categories.
+__________________________
+
 
 
-### cmc_get_options()
+### cmc_get_options
+
 This function returns the applicable options for a specific customer meta category field that is passed to the function and specified by the $identifier.
```

Comment:
Always code snippets in the text must be in block code
I will be happy to consider my opinion. -> $identifier -> `$identifier`